pkg/datasource.go

```diff
@@ -206,6 +243,21 @@ Error:
 	}, nil
 }
 
+func getDeploymentType(s *backend.DataSourceInstanceSettings) (string, error) {
```

Comment:
I don't think there is a good reason to differentiate between "deployment types". The client already knows about all valid scenarios and produces helpful error messages. All of the connection options map to `edgedb.Options` except for instance name. If there is an instance name use `edgedb.CreateClientDSN()` otherwise use `edgedb.CreateClient()`.

We probably do want the UI component to have validation and maybe a way of context switching for cloud vs self hosted, but we don't need to worry about that right now.